ansible_collections/arista/avd/roles/eos_designs/python_modules/base/ip_name_server.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.filter.natural_sort import natural_sort
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+from .utils import UtilsMixin
+
+
+class IpNameServersMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def ip_name_servers(self) -> list | None:
+        """
+        ip_name_servers set based on "dns_settings" data-model or the deprecated "name_servers" input data-model.
+
+        Servers in the new data model may have management VRFs dynamically set.
+        """
+        dns_settings_servers = get(self._hostvars, "dns_settings.servers", default=[])
+        ip_name_servers = self.get_deprecated_name_servers()
+
+        if not dns_settings_servers:
+            return ip_name_servers or None
+
```

Comment:
if we reach here and `ip_name_servers` is not None, we could print a warning to the user

Reply:
No this is to catch the legacy data model alone.